src/App.tsx

```diff
@@ -86,13 +86,20 @@ export const ModalPortal: React.ComponentType = ({ children }) => {
 };
 
 const App: React.FC = () => {
+  const { configureEgoPublicKey } = useAuth();
+
   const client = new ApolloClient({
     uri: process.env.REACT_APP_RDPC_GATEWAY,
     cache: new InMemoryCache()
   });
 
   const globalState = useInitialAppContextState();
 
+  useEffect(() => {
+    configureEgoPublicKey();
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
```

Comment:
this feels a little out of place here... since `configureEgoPublicKey` should really only be called once, it seems a bit odd to be accessible by anything through useAuth... Does it make sense to have the logic built into the `AuthProvider` and have wrapped here instead? The `AuthProvider` wrapper `App` in `index.ts` also feels a little out of place, because it's part of application logic...

Also, I feel like it makes sense to have a simple loading state for it too, cause if it fails to fetch, nothing will work anyway right? Just some idea, what do you think?

Reply:
Yeah, the only reason I have the `AuthProvider` wrapper in `index.tsx` instead of `App.tsx` is because I need immediate access to some of the values it provides (isLoggedIn, for example). Wrapping it in the provider at the same level as it gets used seems to cause issues, so I wrapped it one level up.

Should I create an intermediate layer, like `AppWrapper` or something along those lines to move that out of `index.tsx`?

I also initially had the `configureEgoPublicKey` step being called in a `useEffect` inside the Auth provider itself, but for some reason it wasn't getting run before trying to render the rest of the page which checks things like isValidJwt etc that relied on the ego public key being configured. 😅 this was all so much easier with the hardcoded public key env var hahaha

The loading state is a really good idea, definitely need to implement that and handle the case where the ego public key fetch fails because I just tested that and it causes things to break :P

Any suggestions for how to handle the rest of that?

Reply:
I think the easiest thing to do is just something like this, to avoid having to come up with another name: `export default () => <AuthProvider><App/></AuthProvider>`

As for the failure case, I guess just render a special message saying "Ego is unavailable for authentication at the moment" or some kind of message like that... pretty simple thing I think

Reply:
@hlminh2000 you are a genius!